src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/instance_blackrock_depths.cpp

```diff
@@ -131,13 +140,26 @@ class instance_blackrock_depths : public InstanceMapScript
         uint32 TombTimer;
         uint32 TombEventCounter;
 
+        void UpdateMoira(Creature* moira)
+        {
+            InstanceMap::PlayerList const& players = instance->GetPlayers();
+
+            for (Map::PlayerList::const_iterator i = players.begin(); i != players.end(); ++i)
+                if (Player * player = i->GetSource())
+                    if ((player->GetTeamId() == TEAM_ALLIANCE && !player->IsActiveQuest(QUEST_THE_PRINCESS_SURPRISE))
+                        || (player->GetTeamId() == TEAM_HORDE && !player->IsActiveQuest(QUEST_THE_PRINCESS_SAVED)))
+                        return;
+
+            moira->UpdateEntry(NPC_PRIESTESS_THAURISAN);
```

Comment:
`NPC_PRIESTESS_THAURISSAN`

(or `NPC_HIGH_PRIESTESS_OF_THAURISSAN` if you want to be precise)